### ISSUE TITLE
broker crash in `prepare_sched_status_payload`

### ISSUE THREAD

**Issue by grondo** [flux-framework/flux-core#5868]
Flux was segfaulting just after startup in the resource module `prepare_sched_status_payload()`. 
I was able to run the broker under valgrind, which dumped this error on the first `flux resource list` after startup:

```
==180091== Thread 12:
==180091== Invalid read of size 4
==180091==    at 0x1BFE7414: rank_hash_key_cmp (rlist.c:45)
==180091==    by 0x1BFF4027: s_item_lookup (zhashx.c:332)
==180091==    by 0x1BFF4656: fzhashx_lookup (zhashx.c:430)
==180091==    by 0x1BFE78F9: rank_hash_lookup (rlist.c:66)
==180091==    by 0x1BFE78F9: rlist_find_rank (rlist.c:176)
==180091==    by 0x1BFE78F9: rlist_mark_state (rlist.c:2312)
==180091==    by 0x1BFEB84C: rlist_mark_down (rlist.c:2328)
==180091==    by 0x1BFE47B8: mark_down.part.0 (status.c:105)
==180091==    by 0x1BFE4C5D: mark_down (status.c:100)
==180091==    by 0x1BFE4C5D: create_rlist (status.c:283)
==180091==    by 0x1BFE4C5D: prepare_sched_status_payload (status.c:312)
==180091==    by 0x1BFE4C5D: sched_status_continuation (status.c:382)
==180091==    by 0x4E99A52: ev_invoke_pending (ev.c:3770)
==180091==    by 0x4E9DAF7: ev_run (ev.c:4190)
==180091==    by 0x4E9DAF7: ev_run (ev.c:4021)
==180091==    by 0x4E6C7BE: flux_reactor_run (reactor.c:124)
==180091==    by 0x1BFDE20D: mod_main (resource.c:469)
==180091==    by 0x411DEE: module_thread (module.c:208)
==180091==  Address 0x25096840 is 16 bytes inside a block of size 48 free'd
==180091==    at 0x4C3AC2B: free (vg_replace_malloc.c:974)
==180091==    by 0x1BFE557D: rnode_destroy (rnode.c:48)
==180091==    by 0x1BFE557D: rnode_destroy (rnode.c:41)
==180091==    by 0x1BFE745F: rn_free_fn (rlist.c:124)
==180091==    by 0x1BFF603E: fzlistx_delete (zlistx.c:408)
==180091==    by 0x1BFE852D: rlist_remove_rank (rlist.c:401)
==180091==    by 0x1BFE852D: rlist_remove_ranks (rlist.c:411)
==180091==    by 0x1BFE4C45: create_rlist (status.c:280)
==180091==    by 0x1BFE4C45: prepare_sched_status_payload (status.c:312)
==180091==    by 0x1BFE4C45: sched_status_continuation (status.c:382)
==180091==    by 0x4E99A52: ev_invoke_pending (ev.c:3770)
==180091==    by 0x4E9DAF7: ev_run (ev.c:4190)
==180091==    by 0x4E9DAF7: ev_run (ev.c:4021)
==180091==    by 0x4E6C7BE: flux_reactor_run (reactor.c:124)
==180091==    by 0x1BFDE20D: mod_main (resource.c:469)
==180091==    by 0x411DEE: module_thread (module.c:208)
==180091==    by 0x50DB1C9: start_thread (in /usr/lib64/libpthread-2.28.so)
==180091==  Block was alloc'd at
==180091==    at 0x4C3D1C3: calloc (vg_replace_malloc.c:1554)
==180091==    by 0x1BFE571C: rnode_new (rnode.c:183)
==180091==    by 0x1BFE5855: rnode_create_children (rnode.c:256)
==180091==    by 0x1BFE92B4: rlist_append_rank (rlist.c:879)
==180091==    by 0x1BFE92B4: rlist_append_ranks (rlist.c:927)
==180091==    by 0x1BFE92B4: rlist_append_rank_entry (rlist.c:962)
==180091==    by 0x1BFE92B4: rlist_from_json (rlist.c:1225)
==180091==    by 0x1BFE4C29: create_rlist (status.c:276)
==180091==    by 0x1BFE4C29: prepare_sched_status_payload (status.c:312)
==180091==    by 0x1BFE4C29: sched_status_continuation (status.c:382)
==180091==    by 0x4E99A52: ev_invoke_pending (ev.c:3770)
==180091==    by 0x4E9DAF7: ev_run (ev.c:4190)
==180091==    by 0x4E9DAF7: ev_run (ev.c:4021)
==180091==    by 0x4E6C7BE: flux_reactor_run (reactor.c:124)
==180091==    by 0x1BFDE20D: mod_main (resource.c:469)
==180091==    by 0x411DEE: module_thread (module.c:208)
==180091==    by 0x50DB1C9: start_thread (in /usr/lib64/libpthread-2.28.so)
==180091==    by 0x66EFE72: clone (in /usr/lib64/libc-2.28.so)
```

The problem appears to be in librlist `rlist_mark_down()` after `rlist_remove_ranks()`, however it is mysterious what exactly was causing this that hasn't occurred anywhere else.

**Comment by grondo** [comment]
The problem actually went away while I was looking at it, which makes me think the issue could be reproduced by some combination of exclude and down ranks...

**Comment by grondo** [comment]
Fixed by #5870
